app/services/group/seed.rb

```diff
@@ -0,0 +1,53 @@
+module Group
+  class Seed
+    attr_reader :status
+
+    def initialize(tenant)
+      @user = ManageIQ::API::Common::Request.current.user
+      validate(tenant)
+    end
+
+    def validate(tenant)
+      account_number = ManageIQ::API::Common::Request.current.identity['identity']['account_number']
+      raise Catalog::NotAuthorized if account_number != tenant.external_tenant
+      raise Catalog::NotAuthorized unless @user.org_admin?
+
+      true
+    end
+
+    def tenant
+      @user.tenant
+    end
+
+    def code(status)
+      @status = status
+    end
+
+    def process
+      lookup_groups
+      cache_tenant
+      @seeded.data.present? ? code(204) : run_seeding
+      self
+    end
+
+    private
+
+    def run_seeding
+      seeded = RBAC::Seed.new(Rails.root.join('data', 'rbac_catalog_seed.yml')).process
+      if seeded
+        RbacSeed.create!(:external_tenant => @user.tenant)
+        code(200)
+      end
+    end
+
+    def cache_tenant
```

Comment:
@syncrou How is this caching? Maybe it should be mark_as_seeded. Caching conveys a different meaning (shared memory, locks, multiple processes accessing the data).